File: CS_LIB_ALPHA/Extensions/SystemClass.cs
using System;
using System.Collections;
using System.Collections.Generic;
using System.Diagnostics;
using System.Linq;
using System.Xml.Linq;
using System.Runtime.InteropServices;

    public class FILE
    {
        public IntPtr Pointer { get; set; }

        private Marshal_FILE Values;
        public FILE(string Filename)
            : this(Natives.lept_fopen(Filename, "rb"))
        {
        }

        public FILE(IntPtr IntPTR)
        {
            if (IntPTR == IntPtr.Zero)
                return;
            Pointer = IntPTR;
            Values = new Marshal_FILE();
            Marshal.PtrToStructure(Pointer, Values);
        }

        [StructLayout(LayoutKind.Sequential)]
        private class Marshal_FILE
        {
            public string _ptr;
            public int _cnt;
            public string _base;
            public int _flag;
            public int _file;
            public int _charbuf;
            public int _bufsiz;
            public string _tmpfname;
        }

        public string _ptr
        {
            get { return Values._ptr; }
        }
        public int _cnt
        {
            get { return Values._cnt; }
        }
        public string _base
        {
            get { return Values._base; }
        }
        public int _flag
        {
            get { return Values._flag; }
        }
        public int _file
        {
            get { return Values._file; }
        }
        public int _charbuf
        {
            get { return Values._charbuf; }
        }
        public int _bufsiz
        {
            get { return Values._bufsiz; }
        }
        public string _tmpfname
        {
            get { return Values._tmpfname; }
        }
    }

File: VB_Lib/Extensions/SystemClass.vb
Imports System.Runtime.InteropServices

' ----------------------------------------
' Meistens eine Typedef auf eine Struktur.
' FILE ist eine Systemklasse.
' ----------------------------------------

Public Class FILE
    ' http://www.cplusplus.com/reference/cstdio/fopen/
    Property Pointer As IntPtr
    Private Values As Marshal_FILE

    Sub New(ByVal Filename As String)
        Me.New(LeptonicaSharp.Natives.lept_fopen(Filename, "rb"))
    End Sub

    Sub New(ByVal IntPTR As IntPtr)
        If IntPTR = IntPTR.Zero Then Exit Sub
        Pointer = IntPTR
        Values = New Marshal_FILE
        Marshal.PtrToStructure(Pointer, Values)
    End Sub

    <StructLayout(LayoutKind.Sequential)>
    Private Class Marshal_FILE
        Public _ptr As String
        Public _cnt As Integer
        Public _base As String
        Public _flag As Integer
        Public _file As Integer
        Public _charbuf As Integer
        Public _bufsiz As Integer
        Public _tmpfname As String
    End Class

    ReadOnly Property _ptr As String
        Get
            Return Values._ptr
        End Get
    End Property
    ReadOnly Property _cnt As Integer
        Get
            Return Values._cnt
        End Get
    End Property
    ReadOnly Property _base As String
        Get
            Return Values._base
        End Get
    End Property
    ReadOnly Property _flag As Integer
        Get
            Return Values._flag
        End Get
    End Property
    ReadOnly Property _file As Integer
        Get
            Return Values._file
        End Get
    End Property
    ReadOnly Property _charbuf As Integer
        Get
            Return Values._charbuf
        End Get
    End Property
    ReadOnly Property _bufsiz As Integer
        Get
            Return Values._bufsiz
        End Get
    End Property
    ReadOnly Property _tmpfname As String
        Get
            Return Values._tmpfname
        End Get
    End Property
End Class
